src/routers/datatypeRouters/edges/variants_phenotypes.ts

```diff
@@ -150,90 +187,47 @@ async function getHyperedgeFromVariantQuery (input: paramsFormatType): Promise<a
     phenotypeFilter = `and record._to == 'ontology_terms/${phenotypeFilter}'`
   }
 
-  let queryOptions = ''
-  if (input.pos !== undefined) {
-    queryOptions = 'OPTIONS { indexHint: "region", forceIndexHint: true }'
-  }
-
   let hyperEdgeFilter = getHyperEdgeFilters(input)
-  const variantFilters = getFilterStatements(variantSchema, input)
 
-  if (variantFilters !== '') {
+  if (variantIDs.length > 0) {
     if (hyperEdgeFilter !== '') {
       hyperEdgeFilter = `and ${hyperEdgeFilter}`
     }
-    // variant_id overwrites other fields query on variant
-    if (input._key !== undefined) {
-      query = `
-      LET primaryEdges = (
-        FOR record IN ${variantToPhenotypeSchema.db_collection_name}
-        FILTER record._from == 'variants/${input._key as string}' ${phenotypeFilter}
-        RETURN record._id
-      )
 
-      FOR edgeRecord IN ${variantPhenotypeToStudy.db_collection_name}
-      FILTER edgeRecord._from IN primaryEdges ${hyperEdgeFilter.replaceAll('record', 'edgeRecord')}
-      SORT '_key'
-      LIMIT ${input.page as number * limit}, ${limit}
-      RETURN {
-        'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'edgeRecord._to'},
-        ${getDBReturnStatements(variantPhenotypeToStudy).replaceAll('record', 'edgeRecord')}
-      }
-      `
-    } else {
-      query = `
-      LET primarySources = (
-        FOR record IN ${variantSchema.db_collection_name} ${queryOptions}
-        FILTER ${variantFilters}
-        RETURN record._id
-      )
-
-      LET primaryEdges = (
-        FOR record IN ${variantToPhenotypeSchema.db_collection_name}
-        FILTER record._from IN primarySources ${phenotypeFilter}
-        RETURN record._id
-      )
+    query = `
+    LET primaryEdges = (
+      FOR record IN ${variantToPhenotypeSchema.db_collection_name as string}
+      FILTER record._from IN ['${variantIDs.join('\', \'')}']  ${phenotypeFilter}
+      RETURN record._id
+    )
 
-      FOR edgeRecord IN ${variantPhenotypeToStudy.db_collection_name}
-      FILTER edgeRecord._from IN primaryEdges ${hyperEdgeFilter.replaceAll('record', 'edgeRecord')}
-      SORT '_key'
+    FOR edgeRecord IN ${variantPhenotypeToStudy.db_collection_name as string}
+    FILTER edgeRecord._from IN primaryEdges ${hyperEdgeFilter.replaceAll('record', 'edgeRecord')}
+    SORT '_key'
+    LIMIT ${input.page as number * limit}, ${limit}
+    RETURN {
+      'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'edgeRecord._to'},
+      ${getDBReturnStatements(variantPhenotypeToStudy).replaceAll('record', 'edgeRecord')}
+    }
+    `
+  } else {
```

Comment:
I'm thinking the condition `variantIDs.length > 0` there, so when you do a query for a non-existing variant in input, it could still return some results only based on the `log10pvalue` in input, which is a bit misleading?

Reply:
You are correct. I need to rethink about it.